src/lib/bootloader/systemdboot_widgets.rb

```diff
@@ -27,25 +27,52 @@ def initialize
 
         @minimum = -1
         @maximum = 600
+        @default = 10
       end
 
-      attr_reader :minimum, :maximum
+      attr_reader :minimum, :maximum, :default
 
-      def label
-        _("&Timeout in Seconds")
+      def contents
+        CheckBoxFrame(
+          Id(:cont_boot),
+          _("Finishing systemd-boot menu after a certain amount of time"),
+          false,
+          HBox(
+            IntField(Id(:seconds), _("&Timeout in Seconds"), @minimum, @maximum,
+              systemdboot.menue_timeout.to_i),
+            HStretch()
+          )
+        )
       end
 
       def help
-        _("<p><b>Timeout in Seconds</b>\n" \
+        _("<p>Continue boot process after defined seconds.</p>" \
+          "<p><b>Timeout in Seconds</b>\n" \
           "specifies the time the boot loader will wait until the default kernel is loaded.</p>\n")
       end
 
+      def validate
+        if Yast::UI.QueryWidget(Id(:cont_boot), :Value) &&
+            Yast::UI.QueryWidget(Id(:seconds), :Value) == -1
+          systemdboot.menue_timeout = Yast::ProductFeatures.GetIntegerFeature("globals",
+            "boot_timeout").to_i
+          systemdboot.menue_timeout = @default if systemdboot.menue_timeout <= 0
+          Yast::UI.ChangeWidget(Id(:seconds), :Value, systemdboot.menue_timeout.to_i)
```

Comment:
this should not be in validate. Validate returns true/false if value is valid or not. It should not  do auto corrections. I would simple change in init minimum to `0`.

Reply:
and if needed to use default, do it in `init` method

Reply:
Yes, there has been a better place than validate :-)